ambari-web/app/templates/wizard/downloadMpacks.hbs

```diff
@@ -56,7 +54,11 @@
               </td>
               <td class="col-sm-2">
                 <span data-toggle="tooltip" data-placement="bottom" {{translateAttr title="common.retry"}}>
-                  <button type="button" class="icon-button retry-button" {{bindAttr disabled="mpack.succeeded"}} {{action retryDownload mpack target="controller"}}></button>
+                  {{#if mpack.failed}}
+                  <button type="button" class="icon-button retry-button" {{action retryDownload mpack target="controller"}}></button>
+                  {{else}}
+                  <button type="button" class="icon-button retry-button" disabled="disabled"></button>
+                  {{/if}}
                 </span>
```

Comment:
IMO instead of if else, we can directly bind it
{{bindAttr class=":btn :btn-default :retry-button mpack.failed::disabled"}}

Reply:
Your suggestion inverts the logic. The button must be enabled if mpack.failed is true. I have done it this way because there are three possible states (in progress, succeeded, failed) and the button must only be enabled in one of them. If this were a two-state situation, then I would agree with your approach, but in this case, my way is simpler. It's too bad Handlebars doesn't let us construct more complex conditionals.

Reply:
mpack.failed:disabled would have inverted the logic. mpack.failed::disabled disables it when mpack.failed is false. So as long as the value of mpack.failed is false (or null) the button would remain disabled, doesn't matter it is in-progress or it succeeds. Your if else code above is also doing the exact same thing

Reply:
I tried a couple other options, including binding the disabled attribute directly, but the results were not satisfactory. It looks like in this specific situation, the existing code is a satisfactory solution, if not the most elegant.